internal/gatewayserver/repo/backend.go

```diff
@@ -0,0 +1,186 @@
+package repo
+
+import (
+	"context"
+
+	"github.com/razorpay/trino-gateway/internal/gatewayserver/database/dbRepo"
+	"github.com/razorpay/trino-gateway/internal/gatewayserver/models"
+	"github.com/razorpay/trino-gateway/internal/provider"
+	"github.com/razorpay/trino-gateway/pkg/spine"
+)
+
+type IBackendRepo interface {
```

Comment:
I didn't understand why we are creating these(backend, group, policy, query) interfaces as we have only a single implementation for these and the reqs can also be satisfied using type methods. Will there be any future implementations of these interfaces @utk-spartan ?